soccer/Processor.cpp

```diff
@@ -85,6 +81,16 @@ Processor::Processor(bool sim, bool blueTeam, const std::string& readLogFile)
         &_context, sim, sim ? SimVisionPort : SharedVisionPortSinglePrimary);
     _grSimCom = std::make_unique<GrSimCommunicator>(&_context);
 
+    // Joystick
+    _sdl_joystick_node = std::make_unique<joystick::SDLJoystickNode>();
+    _manual_control_node =
+        std::make_unique<joystick::ManualControlNode>(&_context);
+
+    // Register callbacks
+    _sdl_joystick_node->addCallbacks(_manual_control_node->getCallback(),
+                                     _manual_control_node->getOnConnect(),
+                                     _manual_control_node->getOnDisconnect());
```

Comment:
This is fairly tight coupling between the manual control node and the joystick driver compared to what we have done so far with context. Is this the direction we want to go? An alternate form would be the joystick node pushing a vector of joystick ID's and messages into context that the manual control node consumes every frame

Reply:
The basic idea was to model this as a ROS pub/sub architecture, ie. joystick node publishing joystick messages, manual control node subscribing to those messages. I just came up with this as like a "placeholder" for that when we move over to ROS

Reply:
one thing I'm confused about though is like how this is tightly coupled? The `ManualControlNode` doesn't need to know anything about how the joystick driver works, only that it will call the callbacks when new information is added.

Reply:
They're not like super super tightly coupled, but compared to the context pub/sub model, it's much more coupled. The institutional knowledge of how the context is setup allows much safer editing of one module without the other. Information about how often the callbacks are being called, whether you need to do any multi-thread safety checks, what guarantees there are about which callbacks are called in what order etc, are very important when editing these modules. If this gets added as another model for pub/sub and the institutional knowledge builds assumptions about how data is shared, it becomes much less coupled.

TLDR: If this becomes a common occurrence and pub/sub model alongside the shared memory context model, it's not super coupled. If it doesn't become commonly used and the contract isn't fully defined, it's very coupled

Reply:
Ok, sure. I've changed it to use `context`